### ISSUE TITLE
NAS evaluation is frustratingly hard

### ISSUE THREAD

**Issue by sile** [sile/readings#186]
URL
-----

https://arxiv.org/abs/1912.12522

Abstract
----------

> Neural Architecture Search (NAS) is an exciting new field which promises to be as much as a game-changer as Convolutional Neural Networks were in 2012. Despite many great works leading to substantial improvements on a variety of tasks, comparison between different methods is still very much an open issue. While most algorithms are tested on the same datasets, there is no shared experimental protocol followed by all. As such, and due to the under-use of ablation studies, there is a lack of clarity regarding why certain methods are more effective than others. Our first contribution is a benchmark of 8 NAS methods on 5 datasets. To overcome the hurdle of comparing methods with different search spaces, we propose using a method's relative improvement over the randomly sampled average architecture, which effectively removes advantages arising from expertly engineered search spaces or training protocols. Surprisingly, we find that many NAS techniques struggle to significantly beat the average architecture baseline. We perform further experiments with the commonly used DARTS search space in order to understand the contribution of each component in the NAS pipeline. These experiments highlight that: (i) the use of tricks in the evaluation protocol has a predominant impact on the reported performance of architectures; (ii) the cell-based search space has a very narrow accuracy range, such that the seed has a considerable impact on architecture rankings; (iii) the hand-designed macro-structure (cells) is more important than the searched micro-structure (operations); and (iv) the depth-gap is a real phenomenon, evidenced by the change in rankings between 8 and 20 cell architectures. To conclude, we suggest best practices, that we hope will prove useful for the community and help mitigate current NAS pitfalls. The code used is available at this https URL.

Table of Contents
--------------------

- (1) Introduction
- (2) Related Work
- (3) NAS Benchmark
  - (3-1) Methodology
  - (3-2) Results
- (4) Comparison of Training Protocols
  - (4-1) Methodology
  - (4-2) Results
- (5) Study of DARTS' Search Space
  - (5-1) Distribution of the Random Sampling within DARTS Search Space
  - (5-2) Operations
  - (5-3) Does Changing Seed and Number of Cells Affect Ranking?
- (6) Discussion and Best Practices
- (7) Conclusions
- (A) Appendix
  - (A-1) Methods and Hyperparameters
  - (A-2) Datasets
  - (A-3) Additional Results
    - (A-3-1) Different Traning Protocols for ResNet-50

BibTeX
------

```bibtex
@misc{yang2019nas,
    title={NAS evaluation is frustratingly hard},
    author={Antoine Yang and Pedro M. Esperança and Fabio M. Carlucci},
    year={2019},
    eprint={1912.12522},
    archivePrefix={arXiv},
    primaryClass={cs.LG}
}
```

See Also
----------

- GitHub: https://github.com/antoyang/NAS-Benchmark
